src/pip/_internal/vcs/git.py

```diff
@@ -217,7 +218,7 @@ def resolve_revision(
 
         if sha is not None:
             rev_options = rev_options.make_new(sha)
```

Comment:
`RevOptions.make_new()` seems like unnecessary boilerplate, but I've opted to leave additional (potential) refactoring to another PR.